### PR TITLE
fix(privatek8s): allow Jenkins controller to spawn agents on its own cluster

### PR DESCRIPTION
**Pull request by lemeurherve**
Ref: https://github.com/jenkins-infra/helpdesk/issues/2844

### CODE REVIEW

File: config/ext_jenkins-infra_privatek8s.yaml [hunk 1]
```diff
@@ -50,3 +51,269 @@ controller:
...
+          numExecutors: 0
+          clouds:
+            - kubernetes:
+                containerCapStr: "100"
+                # Kubernetes internal URL to stay within private networks
+                jenkinsUrl: "http://jenkins-infra.jenkins-infra.svc.cluster.local:8080"
+                maxRequestsPerHostStr: "300"
+                webSocket: true
+                name: "kubernetes"
+                namespace: "jenkins-infra-agents"
```
Comment (lemeurherve) [inline config/ext_jenkins-infra_privatek8s.yaml:65]: Changed the namespace and removed the credentialsId, serverUrl and serverCertificate (using the SA instead)

File: clusters/privatek8s.yaml [hunk 2]
```diff
@@ -61,12 +60,16 @@ releases:
       - "../config/ext_jenkins-infra_privatek8s.yaml"
     secrets:
       - "../secrets/config/infra.ci.jenkins.io/jenkins-secrets.yaml"
-  - name: jenkins-infra-additional
-    namespace: jenkins-infra
-    chart: jenkins-infra/jenkins-additional
-    version: 0.0.3
```
Comment (dduportal) [inline clusters/privatek8s.yaml:67]: Good catch. The need for a service account with `cluster-admin` cluster-role (e.g. role applied cluster-wide) was removed in https://github.com/jenkins-infra/kubernetes-management/pull/1861/files#diff-f81821a2c2b64550a4ad4af0a390e2d173489247288e35b23720dceb759274d4 and I never cleaned up.


We'll have to uninstall the resources after merging this PR.

Can you remove the change from this PR though? It deserves another PR (that I'll make)

File: config/ext_jenkins-infra_privatek8s.yaml [hunk 3]
```diff
@@ -50,3 +51,269 @@ controller:
                     description: "Kubeconfig file for privatek8s"
                     scope: SYSTEM
                     secretBytes: "${base64:${KUBECONFIG_PRIVATEK8S}}"
+      agent-settings: |
+        jenkins:
+          numExecutors: 0
+          clouds:
+            - kubernetes:
+                containerCapStr: "100"
+                # Kubernetes internal URL to stay within private networks
```
Comment (dduportal) [inline config/ext_jenkins-infra_privatek8s.yaml:60]:
```suggestion
                #  Controller URL to alllow agent to connect to it. Prefers using the Kubernetes Service as an internal URL.
```

File: config/ext_jenkins-infra_temp-privatek8s.yaml [hunk 4]
```diff
@@ -45,6 +45,276 @@ controller:
                     description: "Kubeconfig file for temp-privatek8s"
                     scope: SYSTEM
                     secretBytes: "${base64:${KUBECONFIG_TEMP_PRIVATEK8S}}"
+      agent-settings: |
```
[the remark this answers is not in the document]
Comment (lemeurherve) [reply config/ext_jenkins-infra_temp-privatek8s.yaml:48]: Oh right, I didn't commited the removal from __common

[the remark this answers is not in the document]
Comment (lemeurherve) [reply config/ext_jenkins-infra_temp-privatek8s.yaml:48]: This is the part I'm not happy with yet, as I need to duplicate this section between the cluster.

File: clusters/privatek8s.yaml [hunk 5]
```diff
@@ -67,6 +67,16 @@ releases:
     version: 0.0.3
     values:
       - ../config/jenkins-infra-additional.yaml
+  - name: jenkins-infra-agents
+    needs:
+      - default/docker-registry-secrets
```
Comment (lemeurherve) [inline clusters/privatek8s.yaml:72]:
```suggestion
```

File: config/ext_jenkins-infra_temp-privatek8s.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (dduportal) [reply config/ext_jenkins-infra_temp-privatek8s.yaml:48]: don't worry too much: we'll throw away the temp-privatek8s conf after so no need to be fancy

**Comment by lemeurherve** [comment]
Kubernetes connection successful:

image

**Review by dduportal — APPROVED** [review]

**Pull request merged**